### ISSUE TITLE
Output from encoder-decoder

### ISSUE THREAD

**Issue by cgbhat1978** [espnet/espnet#5945]
I need to analyze and present visually the training process and the outcome of the encoder-decoder training and compare this with input features. I use 'pytorch' backend process. How can I tap the network to get the output from encoder-decoder setup? Is there a tool already available for comparing the input features and output of encoder-decoder visually?

**Comment by sw005320** [comment]
I'm not very sure exactly what you want, but espnet generates the cross-attention patterns between input and output features during training. You can find some image files in the experimental directory.

**Comment by cgbhat1978** [comment]
Those are at individual utterance level, how can I do this for all utterances at once?

**Comment by sw005320** [comment]
I'm sorry that I could not understand what values you want to extract more exactly.
If you specify them, we may answer your request, but if not, we'd not help more than that.

You can also check this
https://espnet.github.io/espnet/espnet2_tutorial.html#seeing-the-training-status

**Comment by cgbhat1978** [comment]
Thank you. Here's what I need exactly. Considering the transformer model is designed as shown in this figure (https://www.nature.com/articles/s41598-022-12260-y/figures/2) with a set of encoders and decoders, is there a way to extract the embeddings post decoder before the Linear and Softmax layers. Also, is there any tool recommended to visualize these embeddings? This will help me in understanding how the input features have been transformed during training.

**Comment by cgbhat1978** [comment]
Below is the comment from one of the reviewers of my paper, I use ESPnet for data augmentation, which is an extension of https://www.isca-archive.org/interspeech_2022/bhat22_interspeech.pdf
I am unsure how to address this comment, I found the code decoder_layer.py which is getting used but are the embeddings from this layer comparable to the input features (filter bank) at all? Please help.

**In figure 2 (Acoustic features) has been taken as input and after decoder part your extracted data has been received in output layer. A data comparative analysis is required between input features data and after decoder extracted features data using data visualization plot like PCA to show the feature distribution in hypothesis space for model evaluation.**

**Comment by sw005320** [comment]
Thanks for the clarification.
We don't prepare such a function.
You can implement it by yourself.
